Adds Develop with Python Documentation (#2814)
temporalio/documentation: docs/develop/python/index.mdx

## Debugging

The [Debugging feature guide](/develop/python/debugging) covers the various ways to debug your application.

- [Debugging](/develop/python/debugging)

## Schedules

The [Schedules feature guide](/develop/python/schedules) shows how to schedule Workflows, run Workflows on a Cron, and to delay the start of a Workflow.

- [Schedule a Workflow](/develop/python/schedules#schedule-a-workflow)
- [Temporal Cron Jobs](/develop/python/schedules#temporal-cron-jobs)
- [Start Delay](/develop/python/schedules#start-delay)

## Data encryption

The [Data encryption feature guide](/develop/python/data-encryption) provides guidance on how to support compression, encryption, and other data handling by implementing custom converters and codecs.

- [Custom Payload Codec](/develop/python/data-encryption#custom-payload-codec)
- [Payload Conversion](/develop/python/data-encryption#payload-conversion)

## Child Workflows

The [Child Workflows feature guide](/develop/python/child-workflows) shows how to spawn a Child Workflow Execution and handle Child Workflow Events.

- [Start a Child Workflow Execution](/develop/python/child-workflows)

## Continue-As-New

> The [Continue-As-New featuer guide](/develop/python/continue-as-new) shows how to continue the Workflow Execution with a new Workflow Execution using the same Workflow ID.

Reviewer: ```suggestion
The [Continue-As-New feature guide](/develop/python/continue-as-new) shows how to continue the Workflow Execution with a new Workflow Execution using the same Workflow ID.
```

---
id: index
title: Temporal Python SDK Development Documentation
sidebar_label: Python SDK
description: Learn how to use the Temporal Python SDK.
slug: /dev-guide/python
toc_max_heading_level: 4
keywords:
  - dev guide
  - python
  - introduction
tags:
  - dev-guide
  - python
---

> This guide provides a comprehensive overview of the structures, primitives, and features used in [Temporal Application](/temporal#temporal-application) development.

Reviewer: ```suggestion
These Temporal Python SDK feature guides aim to show how to use Temporal features when developing [Temporal Applications](/temporal#temporal-application).
```

Something more like that maybe?

> ## Core Application

Reviewer: ```suggestion
## Core application
```

Still use sentence casing.

> The Core Application section of the Develop with Temporal Python SDK covers the essential concepts and implementation details needed to build and run a Temporal Application.

Reviewer: ```suggestion
The Core Application feature guide shows how to use the basic building blocks of a Temporal Application (Workflows, Activities, and Workers).
```

- [Develop a Basic Workflow](/develop/python/core-application#develop-workflows)
- [Develop a Basic Activity](/develop/python/core-application#develop-activities)
- [Start an Activity Execution](/develop/python/core-application#activity-execution)
- [Run Worker Processes](/develop/python/core-application#run-a-dev-worker)

## Temporal Client

> The Temporal Client section of the Develop with Temporal Python SDK describes how to use the Temporal Client to connect and start a Workflow Execution.

Reviewer: ```suggestion
The Temporal Client feature guide shows how to connect to a Temporal Service and start a Workflow Execution.
```

- [Connect to Development Temporal Service](/develop/python/temporal-clients#connect-to-development-service)
- [Connect a Temporal Client to a Temporal Cluster](/develop/python/temporal-clients#connect-to-a-dev-cluster)
- [Connect to Temporal Cloud](/develop/python/temporal-clients#connect-to-temporal-cloud)
- [Start a Workflow Execution](/develop/python/temporal-clients#start-workflow-execution)

> ## Test Suites

Reviewer: ```suggestion
## Testing suites
```

> The Testing section of the Develop with Temporal Python SDK describes how to test your Temporal Application.

Reviewer: ```suggestion
The [Testing suite feature guide](/develop/python/testing-suite) shows how to setup the testing suite and test Workflows and Activities.
```

- [Test Frameworks](/develop/python/test-suites#test-frameworks)
- [Testing Activities](/develop/python/test-suites#test-activities)
- [Testing Workflows](/develop/python/test-suites#test-workflows)
- [How to Replay a Workflow Execution](/develop/python/test-suites#replay)

> ## Failure Detection

Reviewer: ```suggestion
## Failure detection
```

> The Failure Detection section of the Develop with Temporal Python SDK describes how to configure timeouts and heartbeat intervals for Activities and Workflows.

Reviewer: ```suggestion
The [Failure detection feature guide](/develop/python/failure-detection) shows how your application can detect failures using timeouts and automatically attempt to mitigate them with retries.
```

- [Workflow Timeouts](/develop/python/failure-detection#workflow-timeouts)
- [Set Activity Timeouts](/develop/python/failure-detection#activity-timeouts)
- [Heartbeat an Activity](/develop/python/failure-detection#activity-heartbeats)

## Messages

> The Messages section of the Develop with Temporal Python SDK describes how to use the Temporal Client to send messages to a Workflow Execution.

Reviewer: ```suggestion
The [Application messages feature guide](/develop/python/messages) shows how to send messages to and read the state of Workflow Executions.
```

- [Develop with Signals](/develop/python/messages#signals)
- [Develop with Queries](/develop/python/messages#queries)
- [Develop with Updates](/develop/python/messages#updates)
- [What is a Dynamic Handler](/develop/python/messages#dynamic-handler)

## Cancellation

> The Cancellation section of the Develop with Temporal Python SDK describes how to cancel a Workflow Execution.

Reviewer: ```suggestion
The [Cancellation feature guide](/develop/python/cancellation)shows how to cancel a Workflow Execution.
```
